samples/nrf9160/download/src/main.c

```diff
@@ -137,14 +143,41 @@ static int callback(const struct download_client_evt *event)
 		} else {
 			printk("\r[ %d bytes ] ", downloaded);
 		}
+
+#if CONFIG_SAMPLE_COMPUTE_HASH
+		mbedtls_sha256_update_ret(&sha256_ctx,
+			event->fragment.buf, event->fragment.len);
+#endif
 		return 0;
 
 	case DOWNLOAD_CLIENT_EVT_DONE:
 		ms_elapsed = k_uptime_delta(&ref_time);
 		speed = ((float)file_size / ms_elapsed) * MSEC_PER_SEC;
-
 		printk("\nDownload completed in %lld ms @ %d bytes per sec, total %d bytes\n",
 		       ms_elapsed, speed, downloaded);
+
+#if CONFIG_SAMPLE_COMPUTE_HASH
+		uint8_t hash[32];
+		uint8_t hash_str[64 + 1];
+
+		mbedtls_sha256_finish_ret(&sha256_ctx, hash);
+		mbedtls_sha256_free(&sha256_ctx);
+
+		for (size_t i = 0; i < ARRAY_SIZE(hash); i++) {
```

Comment:
How about using `bin2hex()` from `util.h`?

Reply:
Ah, thanks this is what I had been looking for 👍🏻